driver-core/src/main/java/com/datastax/driver/core/CBUtil.java

```diff
@@ -264,7 +264,9 @@ public static void writeValue(ByteBuffer bytes, ChannelBuffer cb) {
         }
 
         cb.writeInt(bytes.remaining());
-        cb.writeBytes(bytes.duplicate());
```

Comment:
The change globally is not thread safe because ByteBuffer instances passed to this method may be shared between threads. One example is when we clone request messages: currently we use `SessionManager#makeRequestMessage()` to copy messages, and we reuse the same ByteBuffer instances (not even a shallow copy is made with `duplicate()` there). Until we have a better way to clone request messages (see JAVA-726), and more generally, until we can guarantee that ByteBuffer instances passed to this method are never shared, IMHO we can't remove the call to `duplicate()`.